Multisig/UI/Safe Management/Create Safe/CreateSafeFlow.swift

```diff
@@ -108,21 +108,27 @@ class CreateSafeFlow: UIFlow, ASAuthorizationControllerPresentationContextProvid
     }
 
     func googleLogin() {
-        let loginModel = GoogleWeb3AuthLoginModel(authorizationComplete: {
-            let view = self.factory.safeCreatingViewController()
-            view.onSuccess = {
-                self.stop(success: true)
+        let authorizationComplete = { [weak self] in
+            let view = self?.factory.safeCreatingViewController()
+            view?.onSuccess = {
+                self?.stop(success: true)
             }
-            self.show(view)
-
-        } , keyGenerationComplete: { [weak self] key, email in
+            self?.show(view!)
+        }
+        let keyGenerationComplete = { [weak self] (key, email) in
             self?.storeKeyAndCreateSafe(key: key, email: email, keyType: .web3AuthGoogle)
-        })
+        }
 
-        loginModel.loginWithCustomAuth()
+        if loginModel == nil {
+            loginModel = GoogleWeb3AuthLoginModel()
+        }
+        loginModel?.authorizationComplete = authorizationComplete
```

Comment:
at this point, the `loginModel` is never nil, why is the optional unwrapping here? maybe force unwrap or use a local variable and then assign that to the property